### PR TITLE
ControlGroup: Return state data via GetState() by value where applicable

### PR DESCRIPTION
**Pull request by lioncash**
This makes the API for a few of the control groups a little more flexible by returning by value whenever possible. This gets rid of the need to pass locals entirely in some cases, and in other cases, gets rid of the need to pass pointers to locals to certain functions, allowing the state data instances to be const from the get-go.

This also encloses the returned state in their own `StateData` instances, which makes distinct types, preventing accidental assignment to the wrong variable type at the type system level (as opposed to just returning `ControlState` from `GetState()` instances that only return one piece of data), making it a little more type-safe as well.

tl;dr: It makes the interfaces nicer to use.

### CODE REVIEW

File: Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp [hunk 1]
```diff
@@ -175,28 +177,24 @@ void EmulateDynamicShake(AccelData* const accel, DynamicData& dynamic_data,
...
-  // determine which axis is which direction
-  ud = upright ? (sideways ? 0 : 1) : 2;
-  lr = sideways;
-  fb = upright ? 2 : (sideways ? 0 : 1);
-
-  int sgn[3] = {-1, 1, 1};  // sign fix
+  // Determine which axis is which direction
+  const u32 ud = upright ? (sideways ? 0 : 1) : 2;
+  const u32 lr = sideways;
+  const u32 fb = upright ? 2 : (sideways ? 0 : 1);
```
Comment (BhaaLseN) [inline Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp]: This pattern keeps popping up here and there, sometimes as locals and sometimes as array...perhaps not for this PR, but it might be worth considering to consolidate this somewhere.
